openhands/runtime/utils/runtime_build.py

```diff
@@ -197,6 +200,60 @@ def build_runtime_image(
     docker_build_folder: str | None = None,
     dry_run: bool = False,
     force_rebuild: bool = False,
+) -> str:
+    runtime_image_repo, runtime_image_tag = get_runtime_image_repo_and_tag(base_image)
+
+    generic_runtime_image_name = f'{runtime_image_repo}:{runtime_image_tag}'
+
+    # Prepare build folder only once
+    cur_docker_build_folder = docker_build_folder or tempfile.mkdtemp()
+
+    if not force_rebuild and runtime_builder.image_exists(generic_runtime_image_name):
+        from_scratch_hash = prep_docker_build_folder(
+            cur_docker_build_folder,
+            base_image=generic_runtime_image_name,
+            skip_init=True,
+            extra_deps=extra_deps,
+        )
+    else:
+        from_scratch_hash = prep_docker_build_folder(
+            cur_docker_build_folder,
+            base_image=base_image,
+            skip_init=False,
+            extra_deps=extra_deps,
+        )
+
+    hash_runtime_image_name = f'{runtime_image_repo}:{from_scratch_hash}'
```

Comment:
Because when `skip_init=True`, the returned value is no longer from_scratch hash - so this is not really the `hash_runtime_image_name`, which causes later `image_exists` check to not functioning properly?

Reply:
I have to look at that again, the whole logic is a bit confusing what the actual goal is here.
Not having looked at that code for couple of days is like I forgot everything 😂